src/main/java/com/vmware/vipclient/i18n/messages/api/opt/server/ProductBasedOpt.java

```diff
@@ -6,29 +6,30 @@
 
 import java.util.Map;
 
-import org.json.simple.JSONArray;
-
 import com.vmware.vipclient.i18n.VIPCfg;
 import com.vmware.vipclient.i18n.messages.api.opt.BaseOpt;
 import com.vmware.vipclient.i18n.messages.api.opt.Opt;
 import com.vmware.vipclient.i18n.messages.api.url.URLUtils;
 import com.vmware.vipclient.i18n.messages.api.url.V2URL;
 import com.vmware.vipclient.i18n.messages.dto.BaseDTO;
 import com.vmware.vipclient.i18n.util.ConstantsKeys;
+import org.json.simple.JSONArray;
 
+@Deprecated
 public class ProductBasedOpt extends BaseOpt implements Opt {
     private BaseDTO dto = null;
 
     public ProductBasedOpt(BaseDTO dto) {
         this.dto = dto;
     }
 
-    /*
+    /**
      * get supported components from vip(non-Javadoc)
      * 
      * @see com.vmware.vipclient.i18n.messages.dao.IComponentDao#getComponents()
+     * @Deprecated Replaced by {@link com.vmware.vipclient.i18n.messages.api.opt.server.RemoteComponentOpt#getComponents()}
      */
-    public JSONArray getComponentsFromRemoteVIP() {
+    @Deprecated public JSONArray getComponentsFromRemoteVIP() {
```

Comment:
I see you re-implement getComponents in new classes RemoteComponentOpt and LocalComponentOpt, why not re-implement in this class directly, since this method is not exposed to user, you can change it freely; and in Singleton service side, componentlist API is put under product.

In my PR，I put it this method in ProductOpt, so we need reach agreement about this, or there will be duplicate/conflict code.

Reply:
I saw ProductOpt in your PR and I am okay with it. I did not change the code here because you already have the code (and more) in your PR.  I would like to merge this PR first to g11n-java-client branch, so I do not repeat your code. Then could you take care of changing this when you merge your PR?